### PR TITLE
Add endpoints and objects for handling templates.

### PR DESCRIPTION
**Pull request by FasterSpeeding**
### Summary
Add endpoints and objects for handling templates.

### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: hikari/api/rest.py [hunk 1]
```diff
@@ -4960,3 +4960,274 @@ async def fetch_vanity_url(self, guild: snowflakes.SnowflakeishOr[guilds.Partial
...
+            rate-limits and global rate-limits. In some rare edge cases,
+            however, Discord implements other undocumented rules for
+            rate-limiting, such as limits per attribute. These cannot be
+            detected or handled normally by Hikari due to their undocumented
+            nature, and will trigger this exception if they occur.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
+
+    async def create_guild_from_template(
```
Comment (FasterSpeeding) [inline hikari/api/rest.py:5014]: Missed this doc: TODO

File: hikari/guilds.py [hunk 2]
```diff
@@ -513,6 +513,7 @@ def __str__(self) -> str:
         return self.name
 
 
+@attr_extensions.with_copy
```
Comment (FasterSpeeding) [inline hikari/guilds.py:516]: This is unnecessary, should be inherited

File: hikari/api/rest.py [hunk 3]
```diff
@@ -4960,3 +4964,317 @@ async def fetch_vanity_url(self, guild: snowflakes.SnowflakeishOr[guilds.Partial
         hikari.errors.InternalServerError
             If an internal error occurs on Discord while handling the request.
         """
+
```
Comment (davfsa) [inline hikari/api/rest.py:4967]: `@abc.abstractmethod` are missing from all of these

**Review by davfsa — CHANGES_REQUESTED** [review]
For the rest, LGTM

**Review by davfsa — APPROVED** [review]

**Review by davfsa — APPROVED** [review]

**Pull request merged**